src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -3399,7 +3400,14 @@ int sockinfo_tcp::setsockopt(int __level, int __optname,
 	bool supported = true;
 	bool allow_privileged_sock_opt = false;
 
+	if (m_sock_state <= TCP_SOCK_ACCEPT_READY && __optval != NULL && is_inherited_option(__level, __optname))
```

Comment:
I dont agree with this change.
Lets say we are passing invalid argument. so each establish socket will reset this option during creation.

Reply:
I forgot to remove it. I put it under the next condition.